docs/misc/config.md

```diff
@@ -0,0 +1,131 @@
+    static {
+        Pair<ExampleConfig, ForgeConfigSpec> pair = new ForgeConfigSpec.Builder()
+            .configure(ExampleConfig::new);
+        // Store pair values in some constant field
+    }
+    ```
+
+Each config value can be supplied with additional context to provide additional behavior. Contexts must be defined before the config value is fully built:
+
+Method       | Description
+:---         | :---
+`comment`      | Provides a description of what the config value does. Can provide multiple strings for a multiline comment.
+`translation`  | Provides a translation key for the name of the config value.
+`worldRestart` | The world must be restarted before the config value can be changed.
+
+### ConfigValue
+
+Config values can be built with the provided contexts (if defined) using any of the `#define` methods.
+
+All config value methods take in at least two components:
+* A path representing the name of the variable: a `.` separated string representing the sections the config value is in
+* The default value when no valid configuration is present
+
+The `ConfigValue` specific methods take in two additional components:
+* A validator to make sure the deserialized object is valid
+* A class representing the data type of the config value
+
+```java
+// For some ForgeConfigSpec$Builder builder
+ConfigValue<T> value = builder.comment("Comment")
+  .define("config_value_name", defaultValue);
+```
+
+The values themselves can be obtained using `ConfigValue#get`. The values are additionally cached to prevent multiple readings from files.
+
+#### Additional Config Value Types
+
+* **Range Values**
+  * Description: Value must be between the defined bounds
+  * Class Type: `Comparable<T>`
+  * Method Name: `#defineInRange`
+  * Additional Components:
+    * The minimum and maximum the config value may be
+    * A class representing the data type of the config value
+
+!!! note
+    `DoubleValue`s, `IntValue`s, and `LongValue`s are range values which specify the class as `Double`, `Integer`, and `Long` respectively.
+
+* **Whitelisted Values**
+  * Description: Value must be in supplied collection
+  * Class Type: `T`
+  * Method Name: `#defineInList`
+  * Additional Components:
+    * A collection of the allowed values the configuration can be
+
+* **List Values**
+  * Description: Value is a list of entries
+  * Class Type: `List<T>`
+  * Method Name: `#defineList`, `#defineListAllowEmpty` if list can be empty
+  * Additional Components:
+    * A validator to make sure a deserialized element from the list is valid
+
+* **Enum Values**
+  * Description: An enum value in the supplied collection
+  * Class Type: `Enum<T>`
+  * Method Name: `#defineEnum`
+  * Additional Components:
+    * A getter to convert a string or integer into an enum
+    * A collection of the allowed values the configuration can be
+
+* **Boolean Values**
+  * Description: A `boolean` value
+  * Class Type: `Boolean`
+  * Method Name: `#define`
```

Comment:
Looks like the second-level lists don't work with two-space indentation:

![image](https://user-images.githubusercontent.com/21304337/161408700-191cee4d-5ef2-4deb-9ca0-a02cfda25d41.png)